course6/toyapp_alexs/displaylib/transformation3d.cc

```diff
@@ -98,12 +90,12 @@ Transformation3D Transformation3D::WithScaling(const Vector3D& vector) const
 
 Vector3D Transformation3D::Apply(const Vector3D& vector) const
 {
-	Matrix<4, 1> vMatrix {{
+	Matrix<4, 1> vMatrix {
 		vector.x,
 		vector.y,
 		vector.z,
 		1
-	}};
+	};
```

Comment:
So having such a powerful Matrix, why don't you just implement vector like this:
```
template<typename T, int Size>
Vector = Matrix<Size, 1, T>;
```

Or if you need to add vector-specific operations:
```
template<typename T, int Size>
class Vector : public Matrix<Size, 1, T>
{
   // ...
};
```